### ISSUE TITLE
Incorrect channel order for WS2814A RGBW strip.

### ISSUE THREAD

**Issue by zowper** [Aircoookie/WLED#2646]
### What happened?

I am using a WS2814A RGBW strip.  The datasheet for this strip is here: https://drive.google.com/file/d/1IRLl1YkhRFBB01Im5psSYpEsqsy-9JIV/view

I can successfully get all four channels to light up independently, but they're always in the wrong order (with the white channel at the end: xxxW instead of Wxxx).

This issue has been confirmed by Aircoookie on the Discord server:
https://discord.com/channels/473448917040758787/757254961640898622/961198105573867550

Aircoookie — 04/06/2022
"No solution exists yet, making an issue is a good idea and I will get to it soon :)"

### To Reproduce Bug

When it's set to SK6812 RGBW, the Red is White, the Green is Red, the Blue is Green, and the White is Blue (WRGB).

I've tried setting the RGB to no power and the white to full power on the color selector (which lights up as blue in my case).  Then I went to the options and cycled through each color order dropdown (GRB, RGB, etc.)  They are all blue.

### Expected Behavior

I was expecting the white channel to control the white LED, but it cannot.

### Install Method

Binary from WLED.me

### What version of WLED?

0.12.0

### Which microcontroller/board are you seeing the problem on?

ESP32

### Anything else?

The discussion about this issue begins here:
https://discord.com/channels/473448917040758787/757254961640898622/960751408431378442

A simple fix would be to add an extra option to the dropdown to select between WRGB and RGBW.  :)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by zowper** [comment]
I have updated from 0.12.0 to 0.13.1

I set the RGB to 0% brightness and the white channel to 100% brightness.  The strip appeared as 100% blue.  

I then cycled through all six options in "LED Preferences > Color Order".  

They all showed 100% blue (with RGB at 0% and White at 100%).

**Comment by blazoncek** [comment]
Will be available shortly...
![Screenshot 2022-04-29 at 21 02 38](https://user-images.githubusercontent.com/7722392/166009558-3a501bea-e743-4cbd-a8d4-238952806bed.png)

**Comment by Changchung** [comment]
> Will be available shortly... ![Screenshot 2022-04-29 at 21 02 38](https://user-images.githubusercontent.com/7722392/166009558-3a501bea-e743-4cbd-a8d4-238952806bed.png)

This is nice... I need it so bad... Thanks

**Comment by blazoncek** [comment]
Fixed in #2737

**Comment by Oldnumber9** [comment]
Hello, looking for a little help here.

I do not see the "Swap" option in my WLED interface.

I've udpated to 0.13.3 and am using a QuinLED digi quad with WS2814.

My hardware setup is shown in the screenshot below using SK6812 RGBW.  

I have the same wled interface with no "swap" option in both Android and iOS.

I need to swap the white and green channels.  

Any help appreciated.

![image](https://user-images.githubusercontent.com/114184368/191828581-1ef141d9-1f4a-4c95-a777-ee26777e0deb.png)

**Comment by blazoncek** [comment]
Wait for 0.14.

**Comment by Oldnumber9** [comment]
Thank you blazoncek, I will wait for 0.14

**Comment by Losses** [comment]
@Oldnumber9 
Install firmware from here:
https://github.com/srg74/WLED-wemos-shield/tree/master/resources/experimental/Firmware

and config your stipe like this:
![image](https://user-images.githubusercontent.com/1384036/197389965-9786a5f6-eed5-46e1-8322-a09cdc47f060.png)

I've successfully made one, hope this would help you!

**Comment by Oldnumber9** [comment]
> @Oldnumber9 Install firmware from here: https://github.com/srg74/WLED-wemos-shield/tree/master/resources/experimental/Firmware
> 
> and config your stipe like this: ![image](https://user-images.githubusercontent.com/1384036/197389965-9786a5f6-eed5-46e1-8322-a09cdc47f060.png)
> 
> I've successfully made one, hope this would help you!

Thank you Losses, very much appreciated!

I returned the offending WS2814 strip and replaced it with an SK8612, and am now running happily in production.

I appreciate the pointer, and will try your build next time I am flashing.  

Thanks again.

**Comment by hyd808** [comment]
@Losses 
I am having issues saving the swap value. After changing and clicking save, it does not save. It is the only LED setting that will not save.
Any ideas?

Hardware: QuinLED ESP32 board. (ESP32-WROOM-32E)
Firmware: https://github.com/srg74/WLED-wemos-shield/blob/master/resources/experimental/Firmware/WLED_0.14.0-b0_ESP32.bin

**Comment by blazoncek** [comment]
@hyd808 have you updated to latest build of 0.14?

**Comment by hyd808** [comment]
I believe so:
![image](https://user-images.githubusercontent.com/117015361/198909770-75f8e251-cd33-4c3f-8482-8d0e7f28d521.png)

**Comment by hyd808** [comment]
I believe so:
![image](https://user-images.githubusercontent.com/117015361/198909863-b8a963fd-0d78-4b39-a8f9-3ae9a7e6b31c.png)

**Comment by blazoncek** [comment]
I cannot reproduce. In all cases W swapping is saved successfully.

**Comment by speurhond** [comment]
> @Losses I am having issues saving the swap value. After changing and clicking save, it does not save. It is the only LED setting that will not save. Any ideas?
> 
> Hardware: QuinLED ESP32 board. (ESP32-WROOM-32E) Firmware: https://github.com/srg74/WLED-wemos-shield/blob/master/resources/experimental/Firmware/WLED_0.14.0-b0_ESP32.bin

I have the same problem

**Comment by blazoncek** [comment]
> I have the same problem

Erase flash and flash again using one of the supported methods.

**Comment by Losses** [comment]
@blazoncek Confirmed that the latest build has this issue

**Comment by Robertfj** [comment]
@Losses Is it possible for you to build one for ESP02 aswell?

**Comment by blazoncek** [comment]
One thing of note: If saving LED config is unsuccessful it (usually) means that LED initialisation using new settings failed.
In such case it would be necessary to use debug build and monitor serial output.

**Comment by Robertfj** [comment]
It saved successfully for me when using "WLED_0.14.0-b0_ESP8266.bin" on "NodeMCU V3 Lua CH340G ESP8266"

**Comment by jianming0** [comment]
> @Losses I am having issues saving the swap value. After changing and clicking save, it does not save. It is the only LED setting that will not save. Any ideas?
> 
> Hardware: QuinLED ESP32 board. (ESP32-WROOM-32E) Firmware: https://github.com/srg74/WLED-wemos-shield/blob/master/resources/experimental/Firmware/WLED_0.14.0-b0_ESP32.bin

I too am getting the same issue where the swap values are not saving. Am using an ESP32 on this hardware controller: https://www.athom.tech/blank-1/wled-esp32-music-addressable-led-strip-controller

**Comment by blazoncek** [comment]
Read my post above.
If LED config is not saving, something bad happened during re-initialisation of LED buses. To diagnose that you need debug build and monitor serial output.

**Comment by Losses** [comment]
@blazoncek I noticed that the build on Oct 22 works, but users reported the config won't save after Oct 30, maybe some changes break this feature during these days, hope this could provide some cue about the bug.

**Comment by Losses** [comment]
> Read my post above. If LED config is not saving, something bad happened during re-initialisation of LED buses. To diagnose that you need debug build and monitor serial output.

Is there a debug build for users to provide debug information?

**Comment by blazoncek** [comment]
Channel swapping was introduced in 0.14 on August 18th (on April 30th in my fork) and there were no changes to that code since.

**Comment by Losses** [comment]
> Channel swapping was introduced in 0.14 on August 18th (on April 30th in my fork) and there were no changes to that code since.

That's weird, I'll try to use a debug build and see what will happen.

**Comment by jianming0** [comment]
I managed to get the channel swap working by a weird workaround, it takes a few tries for it to work but here it is:

I deleted the LED outputs in the LED preferences and setup the LED strips again this time with a different GPIO number, then changing back to the correct GPIO number.

I have 2 sets of output from my hardware (GPIO 17 for 200 to 300 and GPIO 18 for 0 to 200), so I'd sometimes key in for GPIO 17 first then re-edit them to put GPIO 18 first then GPIO17.

Sometimes it takes a few tries of deleting and changing the order of what is being added first or just adding one by one, for it to work correctly.
